pysb/generator/bng.py

```diff
@@ -249,14 +251,31 @@ def warn_caller(message):
         module = inspect.getmodule(caller_frame)
     warnings.warn(message, stacklevel=stacklevel)
 
+
+class BngPrinter(StrPrinter):
+    def __init__(self, **settings):
+        super(BngPrinter, self).__init__(settings)
+
+    def _print_Piecewise(self, expr):
+        if type(expr.args[-1][1]) != BooleanTrue:
```

Comment:
Weird, `isinstance` does work for me in some minimal test cases with a `Piecewise` object. Anyway, with a true singleton you can use `is` against the singleton instance:
```python
if expr.args[-1][1] is not sympy.true:
```
Then you can also delete the import of `BooleanTrue` above.